django/forms/extras/widgets.py

```diff
@@ -1,149 +1 @@
-        try:
-            year_val, month_val, day_val = value.year, value.month, value.day
-        except AttributeError:
-            year_val = month_val = day_val = None
-            if isinstance(value, six.string_types):
-                if settings.USE_L10N:
-                    try:
-                        input_format = get_format('DATE_INPUT_FORMATS')[0]
-                        v = datetime.datetime.strptime(force_str(value), input_format)
-                        year_val, month_val, day_val = v.year, v.month, v.day
-                    except ValueError:
-                        pass
-                else:
-                    match = RE_DATE.match(value)
-                    if match:
-                        year_val, month_val, day_val = [int(v) for v in match.groups()]
-        html = {}
-        choices = [(i, i) for i in self.years]
-        html['year'] = self.create_select(name, self.year_field, value, year_val, choices, self.year_none_value)
-        choices = list(six.iteritems(self.months))
-        html['month'] = self.create_select(name, self.month_field, value, month_val, choices, self.month_none_value)
-        choices = [(i, i) for i in range(1, 32)]
-        html['day'] = self.create_select(name, self.day_field, value, day_val, choices, self.day_none_value)
-
-        output = []
-        for field in _parse_date_fmt():
-            output.append(html[field])
-        return mark_safe('\n'.join(output))
-
-    def id_for_label(self, id_):
-        for first_select in _parse_date_fmt():
-            return '%s_%s' % (id_, first_select)
-        else:
-            return '%s_month' % id_
-
-    def value_from_datadict(self, data, files, name):
-        y = data.get(self.year_field % name)
-        m = data.get(self.month_field % name)
-        d = data.get(self.day_field % name)
-        if y == m == d == "0":
-            return None
-        if y and m and d:
-            if settings.USE_L10N:
-                input_format = get_format('DATE_INPUT_FORMATS')[0]
-                try:
-                    date_value = datetime.date(int(y), int(m), int(d))
-                except ValueError:
-                    return '%s-%s-%s' % (y, m, d)
-                else:
-                    date_value = datetime_safe.new_date(date_value)
-                    return date_value.strftime(input_format)
-            else:
-                return '%s-%s-%s' % (y, m, d)
-        return data.get(name, None)
-
-    def create_select(self, name, field, value, val, choices, none_value):
-        if 'id' in self.attrs:
-            id_ = self.attrs['id']
-        else:
-            id_ = 'id_%s' % name
-        if not self.is_required:
-            choices.insert(0, none_value)
-        local_attrs = self.build_attrs(id=field % id_)
-        s = Select(choices=choices)
-        select_html = s.render(field % name, val, local_attrs)
-        return select_html
+from django.forms.widgets import SelectDateWidget  # NOQA
```

Comment:
Yes, you're right. I now understand the usage of stacklevel=2 in `django/forms/extras/__init__.py`. I think `stacklevel=2` can stay as it is, but it would be good to add a comment to explain its usecase.

Without stacklevel, the warning location will be wrong for `from django.forms.extras.widgets import SelectDateWidget`:

```
/home/berker/projects/django/django/forms/extras/__init__.py:12: RemovedInDjango21Warning: django.forms.extras is deprecated. You can find SelectDateWidget in django.forms.widgets instead.
  RemovedInDjango21Warning)
```

With stacklevel=2,

```
x.py:1: RemovedInDjango21Warning: django.forms.extras is deprecated. You can find SelectDateWidget in django.forms.widgets instead.
  from django.forms.extras.widgets import SelectDateWidget  # NOQA
```